src/components/dropdownMenu/DropdownMenuItem.tsx

```diff
@@ -0,0 +1,82 @@
+import { useDevice } from "../App";
+
+const MenuItemContent = ({
+  icon,
+  shortcut,
+  label,
+}: {
+  icon?: JSX.Element;
+  shortcut?: string;
+  label: string;
+}) => {
+  const device = useDevice();
+  return (
+    <>
+      <div className="dropdown-menu-item__icon">{icon}</div>
+      <div className="dropdown-menu-item__text">{label}</div>
+      {shortcut && !device.isMobile && (
+        <div className="dropdown-menu-item__shortcut">{shortcut}</div>
+      )}
+    </>
+  );
+};
+
+const DropdownMenuItem = ({
+  icon,
+  onClick,
+  children,
+  dataTestId,
+  shortcut,
+  className = "",
+  link,
+  style,
+}: {
+  icon?: JSX.Element;
+  onClick?: () => void;
+  children: React.ReactNode;
+  dataTestId?: string;
+  shortcut?: string;
+  className?: string;
+  link?: string;
+  style?: React.CSSProperties;
+}) => {
+  const classNames = `dropdown-menu-item ${className}`.trim();
+
+  if (typeof children === "string") {
```

Comment:
checking for `=== "string"` the prevents host app rendering anything even slightly more complex, like a `<Fragment>` or `<span>`. Are we using the fallback render output anywhere at all?

Reply:
Yes I am falling back to `div` in case the type os not a string https://github.com/excalidraw/excalidraw/blob/2b7fb4091c4568a1fc7c03cbdea2574ce03bce1a/src/components/dropdownMenu/DropdownMenuItem.tsx#L74

Reply:
It's so that host can render anything in `MenuItem` and not just be limited to `button` or `link`

Reply:
Yeah, but we can't start breaking the component contract if you're rendering something tiny different from a plain string inside the item.

We also don't want to be highlighting the item if it's not really a list item, as we're doing with the `<LanguageList/>` item.

So I'm wondering if `DropdownMenu` is the right choice here, and whether we don't want a plain `Popover`.

Right now this is an internal implementation as we don't expose the DropdownMenu to users, so we can refactor later. But we'll need to come up with a public API for the menu items so that users can either use an item that should act as a list item (what `DropdownMenuItem` does right now when you supply a plain string) and when it should.. do almost nothing at all I suppose (I'm not sure what Menu styling/behavior we want to retain on items like our LanguageList item)?

---

I think for now we'll tell users that if they want to use items other than `MainMenu.Item`, then they should use `MainMenu.Group` or whatever they want. And we'll document what classes they should use to maintain padding/gaps/etc. correctly.

Reply:
I'm thinking it over and we really should go with the smallest API surface area we can now. Or not necessarily, but try not to be too generic because we don't really know what we're building yet. What we need to support.

So let's just expose the things we use ourselves. And if we think our use case is too specific, let's not export it.

And even then we'll end up breaking great many things anyway.

Reply:
I didn't get it, what's the worry in letting users render whatever they want inside the menu item and not limit it to just a button/link in the main menu? 
I think we should give the flexibility to the host to render any valid children inside the menu items and once we start / if we start supporting more elements eg check box then users can use those but until then its ok for them to render

As per deciding whether to render a button or link, I think we should have `MenuItem.link` instead if the user wants to render a link and not decide based on `onClick` or `link` property.

>  think for now we'll tell users that if they want to use items other than MainMenu.Item, then they should use MainMenu.Group or whatever they want. And we'll document what classes they should use to maintain padding/gaps/etc. correctly.

I don't think documenting classes is a good idea, those are implementation details and the host should never depend on them or need to know about them.


> I'm thinking it over and we really should go with the smallest API surface area we can now. Or not necessarily, but try not to be too generic because we don't really know what we're building yet. What we need to support.

Do you mean reintroduce filtering ?

Reply:
> I didn't get it, what's the worry in letting users render whatever they want inside the menu item and not limit it to just a button/link in the main menu? I think we should give the flexibility to the host to render any valid children inside the menu items and once we start / if we start supporting more elements eg check box then users can use those but until then its ok for them to render
> 
> [...]
>
> Or to simplify more if we don't want to fallback, we can export `Menu.Item.Custom` which would add the wrapper and rest all taken care by host as now.

That's what I'm saying. Let users render what they want. Current implem doesn't allow that. We're rendering three different DOM types and we're ignoring the props on the fallback.

So, yes, let's split up, but a flat structure:

- `MainMenu.Item` or `MainMenu.ItemButton` or `MainMenu.Button`
- `MainMenu.ItemLink` or just `MainMenu.Link`
- `MainMenu.ItemCustom` or just `MainMenu.Custom`

Reply:
I think we should have a default item  so user don't have to think what to render and thats followed across all API's even Radix so I think buttons should be the default one
So Menu.Item we should keep but it should render button
`Menu.Item.Link` => Link
`Menu.Item.Custom` => custom

Reply:
After discussion going ahead with `Menu.ItemLink` and `Menu.ItemCustom`